### PR TITLE
AG-11950 Fix alignment of legend items

### PR DESCRIPTION
**Pull request by olegat**
https://ag-grid.atlassian.net/browse/AG-11950

### CODE REVIEW

**Comment by olegat** [comment]
Note: I've manually checked the update snapshots. They seem OK. The only change is that X-translation of some legend items is getting shifted, which is what's expected.

**Comment by olegat** [comment]
**Updated snapshot diff:**
 [Failed Tests.html.zip](https://github.com/user-attachments/files/15956160/Failed.Tests.html.zip)

File: packages/ag-charts-community/src/chart/legend.ts [hunk 1]
```diff
@@ -501,7 +501,23 @@ export class Legend extends BaseProperties {
...
-    private updateMarkerLabel(markerLabel: MarkerLabel, datum: CategoryLegendDatum): number {
+    private calcMarkerWidth(): number {
+        // AG-11950 Calculate the length of the longest legend symbol to ensure that the text / symbols stay aligned.
+        let result: number = 0;
+        const { showSeriesStroke, marker: itemMarker, line: itemLine } = this.item;
+        this.itemSelection.each((_, datum) => {
+            datum.symbols.forEach((symbol) => {
+                const markerEnabled = this.item.marker.enabled ?? (showSeriesStroke && (symbol.marker.enabled ?? true));
+                const lineEnabled = symbol.line && showSeriesStroke;
+                const lineLength = lineEnabled ? itemLine.length ?? 25 : 0;
```
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/legend.ts]: Nit: Maybe we should factor out a constant for the fallback of `25` here and  line 546?

Comment (olegat) [reply packages/ag-charts-community/src/chart/legend.ts]: Moved duplicate code into a new method `calcSymbolsLengths`

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**